app/assets/javascripts/discourse/app/components/post-list/item/details.gjs

```diff
@@ -1,65 +1,93 @@
 import Component from "@glimmer/component";
 import { hash } from "@ember/helper";
-import { htmlSafe } from "@ember/template";
 import PluginOutlet from "discourse/components/plugin-outlet";
+import TopicStatus from "discourse/components/topic-status";
 import categoryLink from "discourse/helpers/category-link";
 import getURL from "discourse/lib/get-url";
 import { prioritizeNameInUx } from "discourse/lib/settings";
 import { i18n } from "discourse-i18n";
 
 export default class PostListItemDetails extends Component {
+  get url() {
+    return this.args.urlPath
+      ? this.args.post[this.args.urlPath]
+      : this.args.post?.url;
+  }
+
+  get showUserInfo() {
+    if (this.args.showUserInfo !== undefined) {
+      return this.args.showUserInfo && this.args.user;
+    }
+
+    return this.args.user;
+  }
+
+  get topicTitle() {
+    return this.args?.titlePath
+      ? this.args.post[this.args.titlePath]
+      : this.args.post?.title;
+  }
+
   get titleAriaLabel() {
-    return (
-      this.args.titleAriaLabel ||
-      i18n("post_list.aria_post_number", {
-        title: this.args.post.title,
+    if (this.args.titleAriaLabel) {
+      return this.args.titleAriaLabel;
+    }
+
+    if (this.args.post.post_number && this.topicTitle) {
+      return i18n("post_list.aria_post_number", {
+        title: this.topicTitle,
         postNumber: this.args.post.post_number,
-      })
-    );
+      });
+    }
   }
 
   get posterName() {
-    if (prioritizeNameInUx(this.args.post.user.name)) {
-      return this.args.post.user.name;
+    if (prioritizeNameInUx(this.args.user.name)) {
+      return this.args.user.name;
     }
-    return this.args.post.user.username;
+    return this.args.user.username;
   }
 
   <template>
     <div class="post-list-item__details">
       <div class="stream-topic-title">
+        <TopicStatus @topic={{@post}} @disableActions={{true}} />
         <span class="title">
-          <a
-            href={{getURL @post.url}}
-            aria-label={{this.titleAriaLabel}}
-          >{{htmlSafe @post.topic.fancyTitle}}</a>
+          {{#if this.url}}
+            <a
+              href={{getURL this.url}}
+              aria-label={{this.titleAriaLabel}}
+            >{{this.topicTitle}}</a>
+          {{else}}
+            {{this.topicTitle}}
```

Comment:
it's good that we removed the risky `htmlSafe` here. But I think we will now need to add `htmlSafe` to some models/routes. The easy way to check is to create a topic with a title like `Testing <brackets>`. It should render exactly like that in the UI for drafts, activity-stream, etc.

If it renders as `Testing &lt;brackets&gt;`, then it means that it's been double-escaped: once on the server (as part of fancy_title), and once by Ember.

In that situation, we should go to wherever the data is loaded, and add the `htmlSafe()` to that piece of JS.

If it's rendering fine everywhere, then great! Nothing to do.

Reply:
> Adding htmlSafe() here would just undo the escaping right?

No, htmlSafe doesn't actually change the string. It just tells ember "I already escaped this, so you don't need to escape it again".

So I think adding htmlSafe to that line is the right call 👍